modules/swagger-codegen/src/main/resources/JavaSpring/beanValidation.mustache

```diff
@@ -1,5 +1,6 @@
 {{#required}}
   @NotNull
-{{/required}}
+{{/required}}{{#isContainer}}{{^isPrimitiveType}}{{^isEnum}}
```

Comment:
Sure.  I tried this approach first myself.  The problem comes from a property which returns an array.  @Valid validates objets, so Hibernate Validation casts to object during validation and a ClassCastException occurs.

This also removes some load on the Hibernate Validation process I am guessing since it doesn't need to visit all properties.